File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlConditionExpression.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { ComparisonOperator } from './ComparisonOperator';
import { comparisonOperatorToSql } from './ComparisonOperatorExtensions';

export class SqlConditionExpression implements ISqlExpression {
    SqlType: SqlType = SqlType.ComparisonCondition;
    Span: TextSpan = new TextSpan();
    Left!: ISqlExpression;
    ComparisonOperator!: ComparisonOperator;
    OperatorSpan: TextSpan = new TextSpan();
    Right!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ConditionExpression(this);
    }

    ToSql(): string {
        return `${this.Left.ToSql()} ${comparisonOperatorToSql(this.ComparisonOperator)} ${this.Right.ToSql()}`;
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlConditionExpression.cs
namespace T1.SqlSharp.Expressions;

public class SqlConditionExpression : ISqlExpression 
{
    public SqlType SqlType { get; } = SqlType.ComparisonCondition;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ConditionExpression(this);
    }

    public required ISqlExpression Left { get; set; }
    public ComparisonOperator ComparisonOperator { get; set; }
    public TextSpan OperatorSpan { get; set; } = new();
    public required ISqlExpression Right { get; set; }

    public string ToSql()
    {
        return $"{Left.ToSql()} {ComparisonOperator.ToSql()} {Right.ToSql()}";
    }
}